RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -544,6 +544,106 @@ class PumpOpsSynchronous {
         }
         return frameData as Data
     }
+    
+    internal func getGlucoseHistoryEvents(since startDate: Date) throws -> ([TimestampedGlucoseEvent], PumpModel) {
+        try wakeup()
+        
+        let pumpModel = try getPumpModel()
+        
+        var events = [TimestampedGlucoseEvent]()
+        
+        // Going to scan backwards in time through events, so event time should be monotonically decreasing.
+        let eventTimestampDeltaAllowance = TimeInterval(hours: 9)
+        
+        let currentGlucosePage = try readCurrentGlucosePage()
+        let startPage = Int(currentGlucosePage.pageNum)
+        let endPage = max(startPage - 1, 0)
+        
+        pages: for pageNum in stride(from: startPage, to: endPage, by: -1) {
+            NSLog("Fetching page %d", pageNum)
+            let pageData: Data
+            
+            do {
+                pageData = try getGlucosePage(UInt32(pageNum))
+            } catch let error as PumpCommsError {
+                if case .unexpectedResponse(let response, from: _) = error, response.messageType == .emptyHistoryPage {
+                    break pages
+                } else {
+                    throw error
+                }
+            }
+            
+            var idx = 0
+            let chunkSize = 256;
+            while idx < pageData.count {
+                let top = min(idx + chunkSize, pageData.count)
+                let range = Range(uncheckedBounds: (lower: idx, upper: top))
+                NSLog(String(format: "GlucosePage %02d - (bytes %03d-%03d): ", pageNum, idx, top-1) + pageData.subdata(in: range).hexadecimalString)
+                idx = top
+            }
+            
+            let page = try GlucosePage(pageData: pageData, pumpModel: pumpModel)
```

Comment:
I searched and didn't see any actual dependency on pumpModel in the data parsing.  Can this be removed?